patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -337,18 +337,34 @@
 +
 +    /**
 +     * Determines if the specific ItemStack can be placed in the specified armor slot.
++     * @deprecated Override {@link #getEquipmentSlot(ItemStack)} to set the armor slot, and use
++     * {@code armorType == EntityLiving.getSlotForItemStack(stack)} to check the armor slot.
 +     *
 +     * @param stack The ItemStack
 +     * @param armorType Armor slot ID: 0: Helmet, 1: Chest, 2: Legs, 3: Boots
 +     * @param entity The entity trying to equip the armor
 +     * @return True if the given ItemStack can be inserted in the slot
 +     */
++    @Deprecated
 +    public boolean isValidArmor(ItemStack stack, EntityEquipmentSlot armorType, Entity entity)
 +    {
 +        return net.minecraft.entity.EntityLiving.func_184640_d(stack) == armorType;
 +    }
 +
 +    /**
++     * <em><strong>MODDERS</strong>: Override this to set a non-default {@link EntityEquipmentSlot}
++     * for an itemstack, but <strong>do not use this to get the {@link EntityEquipmentSlot} for an
++     * itemstack: for that, use
++     * {@link net.minecraft.entity.EntityLiving#getSlotForItemStack(ItemStack)}.</strong></em>
++     *
++     * @param stack the {@link ItemStack}
++     * @return the {@link EntityEquipmentSlot} for the itemstack, or {@code null} to let
++     * the default vanilla logic as per
++     * {@link net.minecraft.entity.EntityLiving#getSlotForItemStack(ItemStack)} decide
++     */
++    public @Nullable EntityEquipmentSlot getEquipmentSlot(ItemStack stack) { return null; }
```

Comment:
Format this code; put annotation in another line in front. You don't need to squeeze the method in one line since it is totally addition, which will not disturb patch application.